ops/services/app_service/metabase/main.tf

```diff
@@ -1,3 +1,73 @@
+    GRANT USAGE ON SCHEMA public TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE facility_device_type TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE device_type TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE organization TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE patient_answers TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE facility TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE patient_link TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE data_hub_upload TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE time_of_consent TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE specimen_type TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE device_specimen_type TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE facility_device_specimen_type TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    GRANT SELECT ON TABLE test_order TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+
+    GRANT SELECT (internal_id, created_at, updated_at, last_seen, is_deleted) ON TABLE api_user
+    TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+    
+    GRANT SELECT (internal_id, created_at, created_by, updated_at, updated_by, is_deleted, provider_id)
+    ON TABLE provider TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+                
+    GRANT SELECT (internal_id, created_at, created_by, updated_at, updated_by, is_deleted, organization_id, employed_in_healthcare,
+    resident_congregate_setting, facility_id, race, gender, ethnicity, lookup_id, role)
+    ON TABLE person TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+                
+    GRANT SELECT (device_specimen_type_id, created_at, created_by, updated_at, updated_by, patient_id, organization_id, 
+    device_type_id, result, facility_id, survey_data, date_tested_backdate, test_order_id, correction_status, 
+    prior_corrected_test_event_id, internal_id, reason_for_correction)
+    ON TABLE test_event TO ${data.azurerm_key_vault_secret.postgres_readonly_user.value};
+  "
+  EOF
+}
+
+variable "app_settings_overrides" {
+  description = "override values for app_settings"
+  default = {}
+}
+
+resource "null_resource" "add_readonly_db_user" {
+  triggers = {
+    always_run = timestamp()
+  }
+
+  provisioner "local-exec" {
+    command    = var.run_readonly_user_create == false ? "echo" : locals.grant_command
+    on_failure = continue
+  }
+
+  depends_on = [
+    azurerm_key_vault_secret.postgres_readonly_user,
```

Comment:
If these are referring to the changes in ops/prod/_data.tf, then they need to be prefixed by `.data` since they're a data source rather than a resource.

Reply:
Also, does this also depend on the `azurerm_app_service.metabase` resource? Or more likely, does the `azurerm_app_service.metabase` resource depend on `grant_command` having been run?

Reply:
Thanks! I added the `.data` prefix. And the `azurerm_app_service.metabase` should depend on this `grant_command` having already been run, so I added a `depends_on null_resource.add_readonly_db_user` to it - please let me know if my syntax looks off.